File: SistemaEstoqueMercado/src/main/java/j0aod3v/sistemaestoquemercado/service/FuncionarioService.java
package j0aod3v.sistemaestoquemercado.service;

import j0aod3v.sistemaestoquemercado.dao.FuncionarioDAO;
import j0aod3v.sistemaestoquemercado.model.Funcionario;
import java.sql.SQLException;
import java.util.List;

public class FuncionarioService {
    private final FuncionarioDAO funcionarioDAO;

    public FuncionarioService() {
        this.funcionarioDAO = new FuncionarioDAO();
    }

    public void cadastrar(Funcionario funcionario) throws SQLException {
        validar(funcionario);
        funcionarioDAO.inserir(funcionario);
    }

    public Funcionario buscarPorId(int id) throws SQLException {
        validarId(id);
        return funcionarioDAO.buscarPorId(id);
    }

    public List<Funcionario> listarTodos() throws SQLException {
        return funcionarioDAO.listarTodos();
    }

    public void atualizar(Funcionario funcionario) throws SQLException {
        validar(funcionario);
        validarId(funcionario.getIdFuncionario());
        funcionarioDAO.atualizar(funcionario);
    }

    public void excluir(int id) throws SQLException {
        validarId(id);
        funcionarioDAO.excluir(id);
    }

    private void validar(Funcionario funcionario) {
        if (funcionario == null) {
            throw new IllegalArgumentException("Funcionário não pode ser nulo.");
        }
        if (funcionario.getMatricula() == null || funcionario.getMatricula().isBlank()) {
            throw new IllegalArgumentException("Matrícula é obrigatória.");
        }
        if (funcionario.getNome() == null || funcionario.getNome().isBlank()) {
            throw new IllegalArgumentException("Nome é obrigatório.");
        }
        if (funcionario.getCargo() == null || funcionario.getCargo().isBlank()) {
            throw new IllegalArgumentException("Cargo é obrigatório.");
        }
    }

    private void validarId(int id) {
        if (id <= 0) {
            throw new IllegalArgumentException("ID inválido.");
        }
    }
}

File: SistemaEstoqueMercado/frontend/services/funcionarioService.ts
import api from './api';
import { Funcionario } from '@/types';

export const funcionarioService = {
  listarTodos: async (): Promise<Funcionario[]> => {
    const response = await api.get<Funcionario[]>('/funcionarios');
    return response.data;
  },

  buscarPorId: async (id: number): Promise<Funcionario> => {
    const response = await api.get<Funcionario>(`/funcionarios/${id}`);
    return response.data;
  },

  cadastrar: async (funcionario: Funcionario): Promise<Funcionario> => {
    const response = await api.post<Funcionario>('/funcionarios', funcionario);
    return response.data;
  },

  atualizar: async (id: number, funcionario: Funcionario): Promise<Funcionario> => {
    const response = await api.put<Funcionario>(`/funcionarios/${id}`, funcionario);
    return response.data;
  },

  excluir: async (id: number): Promise<void> => {
    await api.delete(`/funcionarios/${id}`);
  },
};